classes/range.js

```diff
@@ -29,10 +29,11 @@ class Range {
     // First reduce all whitespace as much as possible so we do not have to rely
     // on potentially slow regexes like \s*. This is then stored and used for
     // future error messages as well.
-    this.raw = range
-      .trim()
-      .split(/\s+/)
-      .join(' ')
+    this.raw = range.replace(/\s+/g, ' ')
```

Comment:
does it make sense to hoist the regex?

Reply:
I did benchmark a the hoisted version prior to submitting this pull request, and it did not have a measurable impact in my environment. Hoisting won't hurt here, though, I'll do that 👍